README.md

```diff
@@ -497,6 +497,23 @@ Each `extra_arguments` block includes an arbitrary name (in the example above, `
 which the extra arguments should be add, a list of `arguments` or `required_var_files` or `optional_var_files` to add.
 With the configuration above, when you run `terragrunt apply`, Terragrunt will call Terraform as follows:
 
+When available, it is preferable to use interpolation functions such as
+[get_terraform_commands_that_need_locking](#get_terraform_commands_that_need_locking) and
+[get_terraform_commands_that_need_vars](#get_terraform_commands_that_need_vars)
+since they provide the complete list of terraform commands that make use of the desired parameter:
+
+```hcl
+terragrunt = {
+  terraform {
+    # Force Terraform to keep trying to acquire a lock for up to 20 minutes if someone else already has the lock
+    extra_arguments "retry_lock" {
+      commands  = "${get_terraform_commands_that_need_locking()}"
```

Comment:
That doesn't really simplify the implementation because we still have to generate a comma separated list of quoted strings and still have to get rid of the surrounding quotes around the declaration itself.

This is the technique used by terraform. When a command returns an array, the result is an array and there is no need to add the brackets.The only advantage would be to allow the user to specify more commands, but since the list is complete, there is no real use.